crates/ruff_python_formatter/src/builders.rs

```diff
@@ -253,6 +253,16 @@ impl<'ast> Format<PyFormatContext<'ast>> for EmptyWithDanglingComments<'_> {
                 self.opening,
                 // end-of-line comments
                 trailing_comments(&self.comments[..end_of_line_split]),
+                // Avoid unstable formatting with
+                // ```python
+                // x = () - (#
+                // )
+                // ```
+                // Without this the comment would go after the empty tuple first, but still expand
+                // the bin op. In the second formatting pass they are trailing bin op comments
+                // so the bin op collapse. Suboptimally we keep parentheses around the bin op in
+                // either case.
+                (!self.comments[..end_of_line_split].is_empty()).then_some(hard_line_break()),
```

Comment:
What's our motivation to keep the trailing end of line comments on the same line as the opening parentheses? 

I prefer

```python
x = () - (
	# comment
)
```

over 

```python
x = () - ( # comment
)
```

Reply:
Turning an end-of-line comment into an own line comment leads to unstable formatting by the two spaces we insert after the line, so we need turn one comment kind into the other atm.

From a style perspective i prefer black with `()  # comment`

Reply:
> Turning an end-of-line comment into an own line comment leads to unstable formatting by the two spaces we insert after the line, 

We shouldn't insert the two empty spaces in that case. We just format them as own line comments.